### PR TITLE
[example] multi gpus training for mag240m

### PR DESCRIPTION
**Pull request by maqy1995**
## Description
the code is similar to https://github.com/dmlc/dgl/blob/master/examples/pytorch/graphsage/train_sampling_multi_gpu.py

related discussion:https://discuss.dgl.ai/t/will-mag240ms-baseline-code-use-multi-gpus-training-bring-efficiency-improvements/1878

Points that may need attention:
1. It may be necessary to perform performance testing again on the DGL server. The first Epoch may be slower because the feature needs to be cached.
2. I did not adjust the parameters (such as learning rate), which may cause the final accuracy rate to change.
3. Maybe the README should be updated.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by BarclayII — APPROVED** [review]
Thanks a lot!
